src/tracing/swapFlowLoggers.ts

```diff
@@ -31,3 +31,16 @@ export function maybeLogFirstSwapAction(analyticsContext: any) {
     })
   }
 }
+
+export function logSwapQuoteRequest(chainId: number) {
+  const hasSetSwapQuote = tracker.hasTimestamp(SwapEventType.FIRST_QUOTE_FETCH_STARTED)
+  const elapsedTime = tracker.setElapsedTime(SwapEventType.FIRST_QUOTE_FETCH_STARTED)
+  sendAnalyticsEvent(SwapEventName.SWAP_QUOTE_FETCH, {
+    chainId,
+    // We only log the time_to_first_quote_request metric for the first quote request of a session.
+    time_to_first_quote_request: hasSetSwapQuote ? undefined : elapsedTime,
+    time_to_first_quote_request_since_first_input: hasSetSwapQuote
+      ? undefined
+      : tracker.getElapsedTime(SwapEventType.FIRST_QUOTE_FETCH_STARTED, SwapEventType.FIRST_SWAP_ACTION),
+  })
+}
```

Comment:
unit testing this might be a better fit than e2e?

(also makes that  ❌  in the checks go to ✅ )

Reply:
yeah, you're right. i'll add unit tests for these in a follow up PR